module/zfs/zfs_znode.c

```diff
@@ -1346,27 +1308,9 @@ zfs_tstamp_update_setup(znode_t *zp, uint_t flag, uint64_t mtime[2],
 {
 	timestruc_t	now;
 
-	ASSERT(have_tx == !(flag & ATTR_ATIME));
```

Comment:
`have_tx` is now entirely unused, I'd suggest removing the argument entirely to avoid any confusion (from the comment too).